atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionLocksManager.java

```diff
@@ -51,27 +70,40 @@ public Optional<ExpiredLocks> getExpiredImmutableTimestampAndCommitLocks(Optiona
         if (expiredLocks.isEmpty()) {
             return Optional.empty();
         }
-        return Optional.of(ImmutableExpiredLocks.of(getExpiredLocksErrorString(commitLocksToken, expiredLocks)));
+        return Optional.of(ImmutableExpiredLocks.of(getExpiredLocksErrorString(toRefresh, expiredLocks)));
     }
 
-    public SummarizedLockCheckResult getExpiredImmutableTimestampAndCommitLocksWithFullSummary(
-            LockToken commitLocksToken) {
-        Preconditions.checkNotNull(
-                commitLocksToken,
-                "commitLocksToken was null, not expected to be in a call to"
-                        + " getExpiredImmutableTimestampAndCommitLocksWithFullSummary",
+    public synchronized SummarizedLockCheckResult getExpiredImmutableTimestampAndCommitLocksWithFullSummary() {
+        Set<LockToken> locksCopy = getLockTokensCopy();
+        Preconditions.checkState(
+                locksCopy.size() >= ((immutableTimestampLock.isPresent() ? 1 : 0) + 1),
+                "Expected at least one other token, other than immutable timestamp lock, not expected to be in a call"
```

Comment:
I feel like this check is just weird, I think it's shaking out from the fact that the read-path refactoring has not been finished, and eventually this wouldn't be here really?

Reply:
Actually, I think I should just impl a NoOpTransactionLocksManager, and force the immutable commit lock to be always here? And that then simplified this codepath?

Reply:
I am going to quickly try:
* Making TransactionLocksManager and interface, implementing a NoOpTransactionLocksManager and injecting that into read-only transactions and the other place: from what I can see, we always either statically decide to grab the lock or not, it's never weirdly dynamic. We provide Optional.empty in the case of read-only transaction for serializable conflict checkign AND genuine read-only transaction for runTaskWithConditionReadOnlyInternal.

Reply:
And now for my pet peeve that I hope we're going to work on some time soon: the pattern of instantiating SnapshotTransaction JUST TO do reads, is just evil because it pollutes the SnapshotTransaction decomp in a lot of ways.

Reply:
Ok, I special cased the commit lock again and this codepath is back to what it was.